### ISSUE TITLE
Scent masking/reduction. 

### ISSUE THREAD

**Issue by WilliamtheCardboardBox** [CleverRaven/Cataclysm-DDA#32561]
**Is your feature request related to a problem? Please describe.**  
Lately, I've noticed that stealth can be quite inconsistent and rather hard to learn from. I've had experiences where monsters from five or six tiles away suddenly hone in on my player character even though I was crouching, behind cover, with weak scent and light step. I would like to suggest more conscious and strategic options the player can go for.

**Describe the solution you'd like**  
**_Player character smell masking_ - **
When the player pulps or butchers a zombie in a bathtub or similar container it will fill with enough blood and guts to allow the player to bathe themselves in it and reduce their scent profile considerably until positive action, such as bathing, is taken. Pulping a zombe in a suitable container would fill that container with gore and allow for multiple uses. If the player butchers a zombie they will be presented with the option to cover themselves in its guts and blood. This would only allow one use. Covering yourself in monster guts would come with trait dependent morale penalty/positive  "Covered in gore." A feature which would benefit from this, which was in discussion a few months ago, is wind and horde scent tracking. 

**_Player home/location scent masking_ -**
Zombie corpses reduce the player's scent trail in a radius which extends by one tile all the way around the corpse. This would allow for the player to strategically drag and place bodies in areas with high monster density, such as windows, and would reduce the annoying "monsters randomly breaking things" moments. This would also help make sense of monsters tracking the player's scent through piles of rotting zombie corpses. 

_"Fun" details_
If the player is standing near a zombie when it explodes or is instantly gibbed then the player will be soaked in gore. In real life, standing next to someone when they are utterly obliterated usually results in their bone fragments and blood being splattered all over you at high speeds.

**Considering context**
> **Scent is not concerning enough/X for this..**
> **Currently, a morale penalty is not worth adding this...**

I've always been disappointed in the lack of monsters which can extensively track the player through scent. The movie which comes to mind is I Am Legend when Will Smith pours chemicals on the steps of his home to cover his scent trail. Lollygagging in a city of evolved murder machines and then just walking back to your home without a care for the trail of information you're leaving should not be safe. And, if the only purpose that scent will serve is for monsters to break your windows or track you through a few houses then it is not in a satisfactory state.

_**Drawbacks of the player covering themselves in gore.**_
Applying gore to yourself would apply filthy to the player themselves and to their clothing. If the player was to remove all of their clothing and then cover themselves in gore to avoid filthying their clothes then this would break intended balance and would just be awkward and unrealistic during gameplay. So, when the player character has the filthy status they will carry all the negative effects of filthy clothing and any clothing they were to put on would have filthy applied to it. In order for the player to rid themselves of this status they would need to wash themselves with a cleaning agent either in a bathtub which they fill with water or at a water tile like a river.

_**Drawbacks of the player lugging rotting corpses around their home.**_
A morale penalty (e.g, "Rotting Corpses")  would be applied to the player if they are near a corpse for a certain amount of time or if they were to commit certain actions near a corpse. Also, the player would receive a flavor text "It smells like death" indicating that they are receiving the morale penalty from the rotting corpses. 

**Why the drawbacks?**
First, realism. Second, this would be a way for the player to reduce the odds of conflict and combat with monsters and without a trade-off of some sort then it would not be balanced. 

**Details on how the player could reduce their scent**
If the player dirties their clothes by covering themselves in guts then it only makes sense that filthy clothes should also _reduce_ the player's scent which they equip from zombies. This would not be as efficient as the player covering themselves in monster guts but it would reduce it enough to have an impact on the player's scent profile. 

**Comments**
I understand that a feature such as this is low priority and may add gameplay flaws or break the flow of the game given the current proposal, but I think a feature which gives the player more control over the monster's awareness to them, such as the player having the option of controlling and maintaining their scent profile, would be a nice detail to the game and future tracking changes or tracking monsters. 

If this idea could be considered acceptable but not in its current state then please feel free to modify it to a more gameplay friendly state.

Also, and sincerely, please criticize it thoroughly so that if it is considered then it is in the best state that it can be!
**Edits**
TL;DR added.
New revisions/ideas will be included here: 
Removed the idea of gore and entrails all over the player going away with time.
Added "Details on how it could work" which includes ideas which I believe would make sense during gameplay.
New section "Drawbacks" added.
**TL;DR** Conscious player management of scent trail Walking Dead style but without the zombie invisibility. Player bathing to remove the filthy affect from the player. This is not pheromones. 

> **After all consideration this idea is unacceptable in its current state and revision may or may not be possible...**

If this is the case then please tell me and I will go over the entire post again and see if there's any way to make it more suitable for gameplay.

**Comment by ZhilkinSerg** [comment]
There is a thing called zombie pheromone:

https://github.com/CleverRaven/Cataclysm-DDA/blob/a24f3b03bf32b89b309257eff989e031de6dd48d/data/json/items/tools.json#L4329

**Comment by WilliamtheCardboardBox** [comment]
> There is a thing called zombie pheromone:
> 
> https://github.com/CleverRaven/Cataclysm-DDA/blob/a24f3b03bf32b89b309257eff989e031de6dd48d/data/json/items/tools.json#L4329

Pheromone is rare and makes zombies friendly. Also, many people have reported going several seasons before encountering one. My suggestion is reduction of the player's scent trail which will go well with overmap scent tracking when that comes around.

**Comment by anothersimulacrum** [comment]
You can get them by dissecting zombies, which are not exactly rare.

**Comment by roaringjohn** [comment]
As far as I know, pheromones is not really collected much, or even crafted that much. They may not be rare, but still. The thing William suggests would also be useful early game, when you're more careful and don't exactly have access to pheromones.

Not to mention the fact this could go well with the creation of zombies that can track you, as evolved zombies.

**Comment by WilliamtheCardboardBox** [comment]
> As far as I know, pheromones is not really collected much, or even crafted that much. They may not be rare, but still. The thing William suggests would also be useful early game, when you're more careful and don't exactly have access to pheromones.
> 
> Not to mention the fact this could go well with the creation of zombies that can track you, as evolved zombies.

Exactly my point. 



> You can get them by dissecting zombies, which are not exactly rare.

The feature which I am suggesting does not make zombies friendly up to a limit of 500 HP but instead reduces the player's scent trail both in-game and on the overmap. I will include a TL;DR and my idea for where this could play in future updates and game-play.

**Comment by anothersimulacrum** [comment]
>  I am suggesting does not make zombies friendly up to a limit of 500 HP

Pheromones don't work like that anymore.

**Comment by kevingranade** [comment]
I'm not clear on this, but these should both come with downsides.
1. Covering yourself in gore and entrails should apply filthiness to you and your clothing, requiring positive action to remove instead of just fading away.
2. Positioning corpses near your home to mask your scent should also  come with significant drawbacks such as impairing morale.

**Comment by WilliamtheCardboardBox** [comment]
> I'm not clear on this, but these should both come with downsides.
> 
> 1. Covering yourself in gore and entrails should apply filthiness to you and your clothing, requiring positive action to remove instead of just fading away.
> 2. Positioning corpses near your home to mask your scent should also  come with significant drawbacks such as impairing morale.

I've added a drawbacks section. Let me know if I need to include or change anything else!

**Comment by Petethegoat** [comment]
I like this idea a lot. Morale penalties and filthiness sound great.

**Comment by WilliamtheCardboardBox** [comment]
> I like this idea a lot. Morale penalties and filthiness sound great.

I'm thinking that the player washing themselves could go great with a hygiene system down the road too.
